src/components/Footer.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+interface FooterProps {
+  todos: Todo[];
+  filter: string;
+  onFilterChange: (newFilter: string) => void;
+  onClearCompleted: () => void;
+}
+
+const Footer: React.FC<FooterProps> = ({
+  todos,
+  filter,
+  onFilterChange,
+  onClearCompleted,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${todos.filter(todo => !todo.completed).length} items left`}
```

Comment:
Move this logic from jsx to the helper variable

Reply:
done